Include/Rover/Range.hpp

```diff
@@ -75,11 +76,90 @@ namespace Rover {
         Interval interval = Interval::CLOSED);
 
       Type generate(Evaluator& evaluator);
+
+    private:
+      Begin m_begin;
+      End m_end;
+      std::function<Type(const Type&)> m_round;
+      Interval m_interval;
+      std::mt19937 m_engine;
+      std::uniform_real_distribution<double> m_distribution;
   };
 
   template<typename BeginFwd, typename EndFwd>
   Range(BeginFwd&&, EndFwd&&, Interval = Interval::CLOSED) ->
     Range<std::decay_t<BeginFwd>, std::decay_t<EndFwd>, void>;
+
+  template<typename BeginFwd, typename EndFwd, typename GranularityFwd>
```

Comment:
Does the granularity generator have to be a class template parameter? If I did not wrap it into the rounding function, I would have to do a class-scope specialization for void.